### PR TITLE
Adding region landing page for registering/signing in from jambonz.org

### PR DESCRIPTION
**Pull request by khanghugo**
For #36 

Functional wise, it is final. Tested on mobile and it feels like it works the way it is supposed to be.

The styling is close to what the issue suggests. I cannot get the icon to align with the text so maybe that is one improvement.

### CODE REVIEW

**Comment by khanghugo** [comment]
Hmmm, it doesn't like that, let me see what I can do.

**Comment by khanghugo** [comment]
It seems like if I do the other way where I put the query directly into the yaml, it will still throw error, let me see if I can change the test file. Only that one error throws.

**Comment by khanghugo** [comment]
It seems like the best way to fix it is to remove some part of the test. It expects an object but gets a string. That string is brewed from the object. 
https://github.com/jambonz/next-static-site/blob/e90b44a3842af71c3a60eaec1b61e0d2d7fcd9ff/cypress/integration/navi.spec.js#L33-L34

These two lines need axing. First, for the `target="_blank"`, it would create a new tab, we don't need that anymore. The line 33 is as described.

**Comment by khanghugo** [comment]
I have undone some of the stuff. Notably the `url` and the `link` thing. `link` name just happens to pop up when I see it so I use it for the rest of other things. I guess that will leave fewer issues.

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:48]: This layout needs to use the new `pink` altStyle for the `Hero` component. You can do that by adjusting the component structure like this:

```js
<Layout siteData={data.site}>
  <Hero data={data.regions.hero} altStyle="pink">
    <Regions data={data.regions} />
  </Hero>
</Layout>
```

We will also need this then added to the `&--pink` altStyle selector in the `_layout.scss` file to make the headline the jambonz pink color:

```sass
&--pink {
    // ...rest of code...
    h1 {
      color: $jambonz;
    }
    // ...rest of code...
```

And I also had to add a `margin-top` to the `regions` selector to readjust the spacing with this layout change:

```sass
.regions {
  text-align: center;
  // this is new
  margin-top: 64px;
```

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:26]: We should be using the `<P>` jambonz-ui component here for correct type styling. The region titles need to be colored the same as their icon. The link text in this case should be the dark copy color and just the external link icon should be the jambonz pink color. The design is attached to the issue for reference.

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:33]: Again this should be using the `<P>` component instead. You can achieve the proper spacing and alignment by adding a className of `i` to the parent `<a>` element. Just wrap region.text in a span. You can see how this works from the styling here:

https://github.com/jambonz/next-static-site/blob/main/src/styles/_typography.scss#L13-L25

There is an example of this on the Pricing page under Additional Services.

**Review by kitajchuk — CHANGES_REQUESTED** [review]
Looking pretty good. I have some feedback and changes requested here. Ping me if you have any questions.

**Comment by khanghugo** [comment]
Hey,

This is really detailed. Thanks for all of the specific stuffs. Let me see what I can do and hope to resolve all of them accordingly.

**Comment by khanghugo** [comment]
I think it looks pretty okay now. Only the padding seems different from the issue but that is from the hero block and I am not sure how I should change it.

**Comment by kitajchuk** [comment]
> I think it looks pretty okay now. Only the padding seems different from the issue but that is from the hero block and I am not sure how I should change it.

Don’t worry about the padding. I think the design file (Figma) might have an older padding that was wider but we tightened it up during initial development. Since you are using the Hero module the padding is correct. I can check and update the Figma designs to match the real world value 🤔. I can do another full review here later this morning 👍.

File: pages/regions.js
Comment (kitajchuk) [reply pages/regions.js:19]: Hey I still think we should remove this `region` attribute since it serves no semantic purpose 👍

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:28]: We should not be doing inline styles like this. There are some classNames for this in the typography SASS file:

https://github.com/jambonz/next-static-site/blob/main/src/styles/_typography.scss#L110-L120

So here you would instead do it like:

```js
<span className={`color-${region.color}`}>{region.title}</span>
```

The way you have it here the colors `blue` and `teal` aren't even the same color as the icons which use variables from the SASS.

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:27]: Please make sure we use double quotes for JSX syntax 🙏

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:34]: No inline styles please. You can create a selector in your regions SASS file that can target this and apply the color using our SASS variables. We should always style in the SASS using our variables so the colors are consistent.

The color you're looking for here is:

https://github.com/jambonz/next-static-site/blob/main/src/styles/_vars.scss#L14

Yes, it is just the hex for black. But, if we ever wanted to change that to a slightly off black tone we can ensure all typography across the web app will adjust properly. Rendering inline styles like this makes things unpredictable.

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:33]: Again—double quotes for JSX please :)

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:22]: So, personally I don't think we need the icon to be clickable since we have the text link with the external icon UI. This also creates a small a11y issue in Lighthouse: "Links do not have a discernible name". To fix this you can add a `title` attribute to each of these icon `<a>` elements. However, I would prefer we just nix them since the clickable UI is clearly marked with the text links 🤔

**Review by kitajchuk — CHANGES_REQUESTED** [review]

**Comment by kitajchuk** [comment]
Hey @khanghugo reviewing your PR made me realize I needed to fix a layout shift I introduced by lazy loading the react-feather icon JavaScript. I've pushed a commit to `main` here:

https://github.com/jambonz/next-static-site/commit/2c705ae588aac0e77b0762a284e66a7e6b3cd3fc

If you rebase your feature branch with `main` you should notice that icon layout shift go away when hard loading the regions page in the browser.

File: pages/regions.js
Comment (khanghugo) [reply pages/regions.js:19]: Is this the one?
https://github.com/jambonz/next-static-site/blob/ee955ff73425159cc1469f4415011c1cfab1a1ba/pages/regions.js#L14-L16

Comment (khanghugo) [reply pages/regions.js:19]: okay so I removed the wrong one before, let me update that.

**Comment by kitajchuk** [comment]
> Okay, this looks really fancy now. But I am trying to make the code consistent with other pages. Others have their own section tag then start with a header block. But because we want the styling to looks good, I have it started with P tag with className of the according header. It looks scuffed for me. What do you think.
> 
> https://github.com/jambonz/next-static-site/blob/2d3006e063e2e397025979a70519995b03399339/pages/regions.js#L23-L29
> 
> versus
> 
> https://github.com/jambonz/next-static-site/blob/2c705ae588aac0e77b0762a284e66a7e6b3cd3fc/pages/why.js#L33-L40

The `<p>` is fine—we don't need this to be a headline. Not sure what you mean by "scuffed"? Anyway, the nature of visual aesthetic vs semantics is such here. We support typographic style by **tag** (eg `<h1>`) and **className** (eg `.h1`) in order to provide a useful style kit. That being said, you can drop the `h6` className. The design calls for the `<p>` font-size but I do think we should shift the weight a touch. We don't currently have an embedded style for a selector like `p strong` so we could leverage that as a useful addition to our typography styles. We do have a generic style for `strong` [here](https://github.com/jambonz/next-static-site/blob/main/src/styles/_typography.scss#L62-L64) but its weight is bold and I think here it would be better to use medium.

Let me consider this for a minute because we'll want the implementation to be intentional and predictably reusable 🤔

**Comment by kitajchuk** [comment]
@khanghugo In the meantime go ahead and rebase this branch with `main` because I've decided to optimize how we load icons in a different way—a better way less error prone:

https://github.com/jambonz/next-static-site/commit/46549e269659d775c7395f3aa88b32fb8030e8e4

This means in the future we'll always have to add new icons to the imported set if we use new ones. But that's fine. It's more intentional and better than loading all the icons and only using a tiny portion.

File: pages/regions.js
Comment (kitajchuk) [inline pages/regions.js:24]: Okay, here is what I am proposing for how we handle this typography. First, go ahead and add a new selector to the typography SASS file that looks like this:

```sass
.med {
  @include font-medium();
}
```

This gives us a new generic className to utilize for these situations. Then based on my other comments here and the notion of utilizing this new classname I would expect to see the regions__title and regions__text markup end up looking like this:

```js
<div className="regions__title">
  <P className={`med color--${region.color}`}>
    {region.title}
  </P>
</div>
<div className="regions__text">
  <P className="med">
    <Link href={href}>
      <a className="i" title={`jambonz ${region.title}`}>
        <span className="regions__text">{region.text}</span>
        <Icon name="ExternalLink" />
      </a>
    </Link>
  </P>
</div>
```

**Review by kitajchuk — CHANGES_REQUESTED** [review]
Okay—I'm feeling like this is the final one! Please rebase with `main` and once these changes are addressed I think we're looking good. I'll take another look once you're done and just wait for approval before we slam the merge button. I'll have some notes about how I would like the merge handled when we get there 👍

**Comment by khanghugo** [comment]
Okay, now this looks really nice and it makes a ton of sense. Can't wait for you to destroy that merge button 👍.

**Review by kitajchuk — APPROVED** [review]
This looks good! Thanks for your patience with the review process. I feel like this really got dialed in great! I am approving the PR. We merge our own PRs around here, so you get to smash that button. My only ask is that you please use the "Squash and merge" option for the button so we get only one commit in the `main` branch 👍 .

When you merge the production site will automatically be deployed on Vercel 🚀 .

Once you've merged feel free to close the Issue 👍 .

**Pull request merged**
